### PR TITLE
drivers: flash: shell: add verify step to cmd_test

### PR DESCRIPTION
**Pull request by heinwessels**
Previously cmd_test could falsely give the indication that the flash driver functioning properly, because the written data isn't validated, and thus not truley tested.

This commit a verify step to ensure the data was written successfully.

Example output:
```
uart:~$ flash test flash-controller@52002000 100000 1000
Erase OK.
Write OK.
Verified OK.
Erase-Write-Verify test done.
```

```
uart:~$ flash test qspi-nor-flash@0 0 1000
Erase OK.
Write OK.
Verification ERROR!
Erase-Write-Verify test done.
```

Signed-off-by: Hein Wessels <heinwessels93@gmail.com>

### CODE REVIEW

File: drivers/flash/flash_shell.c [hunk 1]
```diff
@@ -241,11 +247,23 @@ static int cmd_test(const struct shell *shell, size_t argc, char *argv[])
...
 		if (result) {
-			shell_error(shell, "Write internal ERROR!");
+			shell_error(shell, "Write failed, code %d!", result);
 			break;
 		}
 
 		shell_print(shell, "Write OK.");
+
+		if (flash_read(flash_dev, addr, check_arr, size) < 0) {
+			shell_print(shell, "Verification read ERROR!");
```
Comment (nordicjm) [inline drivers/flash/flash_shell.c:257]: Why not use result and output error as above?

File: drivers/flash/flash_shell.c [hunk 2]
```diff
@@ -241,11 +247,23 @@ static int cmd_test(const struct shell *shell, size_t argc, char *argv[])
...
+			break;
+		}
+
+		if (memcmp(test_arr, check_arr, size) != 0) {
+			shell_error(shell, "Verification ERROR!");
+			break;
+		}
+
+		shell_print(shell, "Verified OK.");
```
Comment (nordicjm) [inline drivers/flash/flash_shell.c:266]: The part below should be updated to be "Erase-Write-Verify test done."

**Comment by heinwessels** [comment]
@nashif @de-nordic Pinging here, because requesting re-review dismisses all other pending reviews.

File: drivers/flash/flash_shell.c [hunk 3]
```diff
@@ -222,6 +224,10 @@ static int cmd_test(const struct shell *shell, size_t argc, char *argv[])
 		return -EINVAL;
 	}
 
+	if (repeat == 0) {
+		repeat = 1;
+	}
+
```
Comment (de-nordic) [inline drivers/flash/flash_shell.c:230]: This should have separate commit as it does not fix what the commit describes.

Comment (heinwessels) [reply drivers/flash/flash_shell.c:230]: It's now in a separate commit with explanation.

**Review by nordicjm — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Comment by heinwessels** [comment]
> @heinwessels In the stabilization phase, we need a bug linked to each PR. If you'd like to merge before release, file a bug and let me know +1

@laurenmurphyx64 this is not blocking, and I don't mind waiting for the next development phase.

**Pull request merged**